mono/metadata/marshal.c

```diff
@@ -4869,8 +4869,8 @@ mono_marshal_set_last_error (void)
 {
 	/* This icall is called just after a P/Invoke call before the P/Invoke
 	 * wrapper transitions the runtime back to running mode. */
-	MONO_REQ_GC_SAFE_MODE;
 #ifdef WIN32
+	MONO_REQ_GC_SAFE_MODE;
```

Comment:
Is this relevant to this patch?

Reply:
@marek-safar I believe without this change the PR will fail some coop/hybrid suspend tests.  But I think that if we had a Windows build with checked mode it would fail.  So it's not the best fix.

@alexischr The issue is that `mono_marshal_set_last_error` is called from at least 3 different contexts: 1. directly from filewatcher.c in GC Unsafe mode; 2. in this PR from a PAL icall from a GC Safe region; 3. as an icall in its own right in GC Unsafe mode from `marshal-ilgen.c`.

It would be better to have separate entry points for each of those uses with their own requirements that call a common utility function.

Reply:
Not sure if my suggestion should be part of this PR or some followup work.

Reply:
@marek-safar @lambdageek this icall does not exist under Windows (or the situation where this is useful, which is runtime syscall abort) so I don't think it is a problem here.

This is needed because in the case the syscall is aborted by the runtime, in which case the thread state will be STATE_BLOCKING_SUSPEND_REQUESTED by the time we get to this call, which is not allowed by `assert_gc_safe_mode`. It may be that that can be added to the allowed modes?

This is just an easy fix on a check that was not necessary in the first place. It fixes the coop test failures.